File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/post/data/PostImagePagerAdapter.java
package com.pplus.prnumberbiz.apps.post.data;

import android.content.Context;
import android.graphics.BitmapFactory;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.pple.pplus.utils.part.info.DeviceUtil;
import com.pplus.prnumberbiz.R;
import com.pplus.prnumberbiz.core.network.model.dto.Attachment;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by KIMSOONHO on 2017-02-01.
 */

public class PostImagePagerAdapter extends PagerAdapter{

    private final LayoutInflater mInflater;
    private Context mContext;
    private List<Attachment> dataList = new ArrayList<>();
    private OnItemClickListener mListener;
    private int width, height;

    public interface OnItemClickListener{

        void onItemClick(int position);
    }

    public void setListener(OnItemClickListener mListener){

        this.mListener = mListener;

    }

    public PostImagePagerAdapter(final Context context, List<Attachment> dataList){

        mContext = context;
        mInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.dataList = new ArrayList<>();
        this.dataList.addAll(dataList);
        width = DeviceUtil.DISPLAY.SCREEN_WIDTH_PIXELS - context.getResources().getDimensionPixelSize(R.dimen.width_144);
        height = (int) (width * 0.75);
    }

    public void setDataList(List<Attachment> dataList){

        this.dataList = new ArrayList<>();
        this.dataList.addAll(dataList);
        notifyDataSetChanged();
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object){

        ((ViewPager) container).removeView((View) object);
    }

    public List<Attachment> getDataList(){

        return dataList;
    }

    @Override
    public int getCount(){

        return dataList.size();
    }

    @Override
    public Object instantiateItem(ViewGroup container, final int position){

        View view = mInflater.inflate(R.layout.item_post_image, container, false);

        ImageView imageView = (ImageView) view.findViewById(R.id.image_post);

        Attachment data = dataList.get(position);

        Glide.with(mContext).load(data.getUrl()).apply(new RequestOptions().fitCenter().placeholder(R.drawable.prnumber_default_img).error(R.drawable.prnumber_default_img)).into(imageView);

        imageView.setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View view){

                if(mListener != null) {
                    mListener.onItemClick(position);
                }
            }
        });

        container.addView(view);
        return view;
    }

    @Override
    public boolean isViewFromObject(View view, Object object){

        return view == object;
    }

    public BitmapFactory.Options getBitmapSize(String fullname){

        BitmapFactory.Options options = new BitmapFactory.Options();
        options.inJustDecodeBounds = true;
        BitmapFactory.decodeFile(new File(fullname).getAbsolutePath(), options);
        return options;
    }
}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/post/data/PostImagePagerAdapter.kt
package com.pplus.prnumberuser.apps.post.data

import android.content.Context
import android.graphics.BitmapFactory
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.bumptech.glide.Glide
import com.bumptech.glide.load.model.GlideUrl
import com.bumptech.glide.request.RequestOptions
import com.pplus.prnumberuser.Const
import com.pplus.prnumberuser.R
import com.pplus.prnumberuser.core.network.model.dto.Attachment
import com.pplus.prnumberuser.databinding.ItemPostImageBinding
import com.pplus.utils.part.info.DeviceUtil
import java.io.File
import java.util.*

/**
 * Created by KIMSOONHO on 2017-02-01.
 */

class PostImagePagerAdapter : PagerAdapter {

    private var mInflater: LayoutInflater
    private var dataList: MutableList<Attachment> = ArrayList()
    private var mListener: OnItemClickListener? = null
    private val width: Int
    private val height: Int

    constructor(context: Context, dataList: List<Attachment>) : super() {
        mInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        this.dataList = ArrayList()
        this.dataList.addAll(dataList)
        width = DeviceUtil.DISPLAY.SCREEN_WIDTH_PIXELS - context.resources.getDimensionPixelSize(R.dimen.width_144)
        height = (width * 0.75).toInt()
    }

    interface OnItemClickListener {

        fun onItemClick(position: Int)
    }

    fun setListener(mListener: OnItemClickListener) {

        this.mListener = mListener

    }


    fun setDataList(dataList: List<Attachment>) {

        this.dataList = ArrayList()
        this.dataList.addAll(dataList)
        notifyDataSetChanged()
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {

        (container as ViewPager).removeView(`object` as View)
    }

    fun getDataList(): List<Attachment> {

        return dataList
    }

    override fun getCount(): Int {

        return dataList.size
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {

        val binding = ItemPostImageBinding.inflate(mInflater, container, false)

        val imageView = binding.imagePost

        val data = dataList[position]

        val id = data.id
        val glideUrl = GlideUrl("${Const.API_URL}attachment/image?id=${id}")
        Glide.with(imageView.context).load(glideUrl).apply(RequestOptions().fitCenter().placeholder(R.drawable.prnumber_default_img).error(R.drawable.prnumber_default_img)).into(imageView)

        imageView.setOnClickListener {
            if (mListener != null) {
                mListener!!.onItemClick(position)
            }
        }

        container.addView(binding.root)
        return binding.root
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {

        return view === `object`
    }

    fun getBitmapSize(fullname: String): BitmapFactory.Options {

        val options = BitmapFactory.Options()
        options.inJustDecodeBounds = true
        BitmapFactory.decodeFile(File(fullname).absolutePath, options)
        return options
    }
}
